test/Microsoft.SqlTools.ServiceLayer.IntegrationTests/Migration/MigrationServiceTests.cs

```diff
@@ -34,7 +38,34 @@ public async Task TestHandleMigrationAssessmentRequest()
                 MigrationService service = new MigrationService();
                 await service.HandleMigrationAssessmentsRequest(requestParams, requestContext.Object);
                 requestContext.VerifyAll();
-            }     
+            }
+        }
+
+        [Test]
+        public async Task TestHandleMigrationGetSkuRecommendationsRequest()
+        {
+            GetSkuRecommendationsResult result = null;
+
+            var requestParams = new GetSkuRecommendationsParams()
+            {
+                DataFolder = Path.Combine("..", "..", "..", "Migration"),
+                TargetPlatforms = new List<string> { "AzureSqlManagedInstance" },
+                TargetSqlInstance = "Test",
+                TargetPercentile = 95,
+                StartTime = new DateTime(2020, 01, 01).ToString("yyyy-MM-dd HH:mm:ss"),
+                EndTime = DateTime.UtcNow.ToString("yyyy-MM-dd HH:mm:ss"),
+                PerfQueryIntervalInSec = 30,
+                ScalingFactor = 1,
+                DatabaseAllowList = new List<string> { "test", "test1" }
+            };
+
+            var requestContext = RequestContextMocks.Create<GetSkuRecommendationsResult>(r => result = r).AddErrorHandling(null);
+
+            MigrationService service = new MigrationService();
+            await service.HandleGetSkuRecommendationsRequest(requestParams, requestContext.Object);
+            Assert.IsNotNull(result, "Get SKU Recommendation result is null");
+            Assert.IsNotNull(result.SqlMiRecommendationResults, "Get MI SKU Recommendation result is null");
+            Assert.AreEqual(result.SqlMiRecommendationResults.First().PositiveJustifications.Count, 6, "Invalid number of positive justifications for MI SKU Recommendation result");
```

Comment:
I would say change this to:
1) use PositiveJustifications concatenated with NegativeJustifications
2) change it so that the combined length is maybe >0 or >1 or >2 or something, instead of equal to exactly 6 (since the number of justifications returned can change depending on the characteristics of the specific source instance, even if we don't change the recommendation logic)

Reply:
My understanding is if we get a negative justification, we will exclude that SKU from final recommended list.

Reply:
That's a good point. In practice you're right, if the SKU recommendation result actually makes it back to the end user then it won't have any negative justifications. Although it's possible that in the future (in cases where we recommend multiple recommendations at a time, for example) we may start to start using it. So I figured I'd mention it now to include it just in case, but it shouldn't matter for now